File: shell/src/main/java/com/crystal/shell/utils/FileUtils.java
package com.crystal.shell.utils;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.RandomAccessFile;

/**
 *
 * date 2020/8/6.
 * description：
 */
public class FileUtils {
    /**
     * 递归删除目录下的所有文件及子目录下所有文件
     *
     * @param path 将要删除的文件目录地址
     */
    public static void delFolder(String path) {
        File delFile = new File(path);
        if (delFile.exists()) {
            File[] files = delFile.listFiles();
            //递归删除目录中的子目录下
            for (File file : files) {
                if (file.isDirectory()) {
                    //递归直到目录下没有文件
                    delFolder(file.getAbsolutePath());
                } else {
                    //删除
                    file.delete();
                }
            }
            delFile.delete();
        }
    }

    /**
     * 写入数据
     *
     * @param file 写入文件地址
     * @param data 需要写入的数据
     */
    public static void wirte(File file, byte[] data) {
        FileOutputStream fos = null;
        try {
            fos = new FileOutputStream(file);
            fos.write(data);
            fos.flush();
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            if (fos != null) {
                try {
                    fos.close();
                    fos = null;
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    public static byte[] getBytes(File dexFile) {
        try {
            RandomAccessFile fis = new RandomAccessFile(dexFile, "r");
            byte[] buffer = new byte[(int) fis.length()];
            fis.readFully(buffer);
            fis.close();
            return buffer;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }


    /**
     * 文件拷贝
     *
     * @param srcFile 原文件
     * @param dstFile 目标文件
     */
    public static void copyFile(File srcFile, File dstFile) {
        if (srcFile == null || !srcFile.exists()) {
            LogUtils.e("Copy the original file does not exist");
            return;
        }
        FileOutputStream fos = null;
        byte[] bytes = FileUtils.getBytes(srcFile);
        try {
            fos = new FileOutputStream(dstFile);
            fos.write(bytes);
            fos.flush();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (fos != null) {
                try {
                    fos.close();
                    fos = null;
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }

    }
}

File: encrypt/src/main/java/com/crystal/encrypt/utils/FileUtils.kt
package com.crystal.encrypt.utils

import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.RandomAccessFile

/**
 * 文件工具类
 * @description
 * @author XiXu
 * on 2023/4/18
 */
object FileUtils {

    /**
     * 递归删除目录下的所有文件以及子目录下所有文件
     */
    fun deleteFolder(path: String) {
        val deleteFile = File(path)
        if (deleteFile.exists()) {
            var listFiles = deleteFile.listFiles()
            //递归删除目录下的子目录
            for (file in listFiles) {
                if (file.isDirectory) {
                    deleteFolder(file.absolutePath)
                } else {
                    file.delete()
                }
            }
            deleteFile.delete()
        }
    }

    /**
     * 写入数据
     */
    fun write(file: File, data: ByteArray) {
        val fos = FileOutputStream(file)
        fos.write(data)
        fos.flush()
        fos.close()
    }

    /**
     * 读取文件为bytes数据
     */
    fun getBytes(dexFile: File): ByteArray {
        val fis = RandomAccessFile(dexFile, "r")
        val buffer = ByteArray(fis.length().toInt())
        fis.readFully(buffer)
        fis.close()
        return buffer
    }

    /**
     * 文件拷贝
     */
    fun copyFile(srcFile: File, dstFile: File) {
        if (!srcFile.exists()) {
            LogUtils.e("unZip :scrFile is null")
            return
        }
        val bytes = getBytes(srcFile)
        val fos = FileOutputStream(dstFile)
        fos.write(bytes)
        fos.flush()
        fos.close()
    }
}